src/pretix/base/datasync/datasync.py

```diff
@@ -0,0 +1,425 @@
+    :return: {
+                "object_type": mapping.external_object_type,
+                "pk_field": pk_field,
+                "pk_value": pk_value,
+                "external_link_href": "https://external-system.example.com/backend/link/to/contact/123/",
+                "external_link_display_name": "Contact #123 - Jane Doe",
+                "...optionally further values you need in mapped_objects for association": 123456789,
+             }
+
+    This method needs to be idempotent, i.e. calling it multiple times with the same input values should create
+    only a single object in the target system.
+
+    Subsequent calls with the same mapping and pk_value should update the existing object, instead of creating a new one.
+    In a SQL database, you might use an "INSERT OR UPDATE" or "UPSERT" statement; many REST APIs provide an equivalent API call.
+    """
+    def sync_object_with_properties(
+            self,
+            pk_field,
+            pk_value,
+            properties: list,
+            inputs: dict,
+            mapping,
+            mapped_objects: dict,
+            **kwargs,
+    ):
+        raise NotImplementedError()
+
+    def sync_object(
+            self,
+            inputs: dict,
+            mapping,
+            mapped_objects: dict,
+    ):
+        logger.debug("Syncing object %r, %r, %r", inputs, mapping, mapped_objects)
+        properties = self.get_properties(inputs, mapping.property_mapping)
+        logger.debug("Properties: %r", properties)
+
+        pk_value = self.get_field_value(inputs, {"pretix_field": mapping.pretix_pk})
+        if not pk_value:
+            return None
+
+        info = self.sync_object_with_properties(
+            pk_field=mapping.external_pk,
+            pk_value=pk_value,
+            properties=properties,
+            inputs=inputs,
+            mapping=mapping,
+            mapped_objects=mapped_objects,
+        )
+        OrderSyncLink.objects.create(
+            order=inputs.get(ORDER), order_position=inputs.get(ORDER_POSITION), sync_provider=self.identifier,
+            external_object_type=info.get('object_type'),
+            external_pk_name=info.get('pk_field'),
+            external_pk_value=info.get('pk_value'),
+            external_link_href=info.get('external_link_href'),
+            external_link_display_name=info.get('external_link_display_name'),
+        )
+        return info
+
+    def sync_order(self, order):
+        if not self.order_valid_for_sync(order):
+            logger.debug("Skipping order %r (not valid for sync)", order)
+            return
+
+        logger.debug("Syncing order %r", order)
+        positions = list(
+            order.all_positions.filter(item__admission=True)
+            .prefetch_related("answers", "answers__question")
+            .select_related(
+                "voucher",
+            )
+        )
+        order.synced_objects.filter(sync_provider=self.identifier).delete()
```

Comment:
Huh, what? We delete the information of the previous sync before we sync again? That sounds weird, so if the sync fails, we lost the connection permanently?